### ISSUE TITLE
Chisel and bits not working on many blocks.

### ISSUE THREAD

**Issue by SimplyLime** [ChiselsAndBits/Chisels-and-Bits#411]
Hey, I am running a server the foolcraft 1.4.5 pack and using chisels and bits along with the extra bits manipulation addon. I find that players straight up cannot chisel and bit blocks from the same mod, and even some vanilla blocks are unchiselable. obsidian for example. when they left click the block it shaves off the bit but immediately comes back. I realize I am 2 versions behind and will have to wait for a pack update but upon looking at the change logs this issue doesn't seem to be addressed. Thanks in advance.

For any bug reports please provide the following details as well,
* MC Version: 1.12
* C&B Version: 14.18
* Do You have Optifine: no

**Comment by AlgorithmX2** [comment]
I was unable to reproduce such an issue, I can chisel obsidian with a diamond chisel without issue, I tested in single player and multiplayer just to make sure.

There must be another mod in the pack interfering with the process, or disabling it via permissions or something. I'm not really sure.

**Comment by SimplyLime** [comment]
It seems to only be on my server. I am using spongeforge, so it’s possible that that is what’s causing it. Should I report it with them?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by AlgorithmX2** [comment]
Do you have any permission mods installed and setup? There were recent patches to C&B to support them. So it might be that either one is reporting things wrong, or there is some issue with that compatibility. I relied on others to test that functionality and they seemed to think that everything was working.

If there isn't anything preventing the actions, we might have to report it to them and see if we can sort out whats going on there.

**Comment by SimplyLime** [comment]
Yes I am using luck perms with spongeforge. I wasn’t aware there were any permission nodes for c&b. I am going to message sponge forge and grief prevention but if you can give any further insight that would be most appreciated. Thanks. :)

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by AlgorithmX2** [comment]
I'm not sure if there are nodes specifically for C&B, but C&B should obey the general access rules for breaking blocks. C&B checks to see if you can break a block before letting you remove bits from it for example.

**Comment by SimplyLime** [comment]
I see, well blocks can definately be broken by players. the more bazaar thing is that it is only certain blocks. which leads me to believe its some sort of bug between sponge and c&b. It even works on some modded blocks but not others from the same mod.

**Comment by AlgorithmX2** [comment]
Seems like there is some confusion somewhere certainly.

We'll probably need more insight from the other involved mods. Where I'm sitting i'm not seeing anything on my end that seems wrong, but then again its hard to see anything wrong with something as mundane as "if not modifiable, then don't modify." This sounds like its gonna probably have a bit more nuance however.

**Comment by ResqDiver1317** [comment]
Out of curiosity, are you running SpongeForge? I just did an update on a few mods on our server and updated my SpongeForge to the most recent version and it broke our ability to chisel on many blocks. I rolled my SpongeForge back to an older version and it's working again.

**Comment by SimplyLime** [comment]
I am actually! Could i get the version number you rolled back to? I figured it was something to do with sponge.

**Comment by ResqDiver1317** [comment]
I'm running spongeforge-1.12.2-2705-7.1.0-BETA-3328

**Comment by ResqDiver1317** [comment]
Sponge is the bane of my existence sometimes

**Comment by SimplyLime** [comment]
yeah for sure haha, how long ago did you try to update? im thinking maybe the new updates might fix this issue because im also a few versions behind now.

**Comment by ResqDiver1317** [comment]
I updated my SpongeForge tonight and broke everything. The version I went back to is several weeks old I think (but seems to work better)

**Comment by ResqDiver1317** [comment]
I've found with Sponge, many times their updates fix things, but break others in the process

**Comment by SimplyLime** [comment]
LOL as you said that I have been rolling back from latest back to 3344. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Shybella** [comment]
I'm having this too and using Spongeforge 3374

**Comment by Shybella** [comment]
When version did this start happening for you all on Spongeforge?

**Comment by SimplyLime** [comment]
After testing it, downgrading to 3328 did infact fix the issue, alternatively there was a recent recommended build of sponge released so you could try that out if you don’t want the older version. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Shybella** [comment]
Thank you, I'm actually using one of the latest builds. I'm reporting this issue on Sponge. If you want to reply here, please do. 

https://github.com/SpongePowered/SpongeForge/issues/2369

**Comment by ResqDiver1317** [comment]
Shybella, I'm not sure what version it started happening on as I don't update my SpongeForge very often unless there is a new recommended version or unless I see in their notes that it is fixing an issue I have been seeing in our modpack (primarily because I often run into issues like this with the updates). All I do know is that the issue occurred somewhere after 3328. Has anyone tested this issue against the latest recommended build (3361) to see if this is still occurring in that version or not?

**Comment by AlgorithmX2** [comment]
Can anyone confirm if this is still an issue with the current builds of both C&B and SpongeForge?

**Comment by TeraNovell** [comment]
@AlgorithmX2 

I tested it again with the following Versions:

Sponge: 1.12.2-2705-7.1.0-BETA-3422
Chisels & Bits: 14.23

And this error is still present:

```[21:52:45] [Server thread/FATAL] [net.minecraft.server.MinecraftServer]: Error executing task
java.util.concurrent.ExecutionException: java.lang.ClassCastException: mod.chiselsandbits.chiseledblock.HarvestWorld cannot be cast to org.spongepowered.common.interfaces.world.IMixinWorld
	at java.util.concurrent.FutureTask.report(Unknown Source) ~[?:1.8.0_181]
	at java.util.concurrent.FutureTask.get(Unknown Source) ~[?:1.8.0_181]
	at net.minecraft.util.Util.func_181617_a(SourceFile:47) [h.class:?]
	at org.spongepowered.common.SpongeImplHooks.onUtilRunTask(SpongeImplHooks.java:293) [SpongeImplHooks.class:1.12.2-2705-7.1.0-BETA-3422]
	at net.minecraft.server.MinecraftServer.redirect$onRun$zjj000(MinecraftServer.java:3972) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:723) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:396) [nz.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526) [MinecraftServer.class:?]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_181]
Caused by: java.lang.ClassCastException: mod.chiselsandbits.chiseledblock.HarvestWorld cannot be cast to org.spongepowered.common.interfaces.world.IMixinWorld
	at net.minecraftforge.common.ForgeHooks.canHarvestBlock(ForgeHooks.java:1689) ~[ForgeHooks.class:?]
	at net.minecraft.block.Block.canHarvestBlock(Block.java:1154) ~[aow.class:?]
	at mod.chiselsandbits.chiseledblock.BlockChiseled.canHarvestBlock(BlockChiseled.java:1139) ~[BlockChiseled.class:?]
	at mod.chiselsandbits.items.ItemChisel.canMine(ItemChisel.java:409) ~[ItemChisel.class:?]
	at mod.chiselsandbits.items.ItemChisel.chiselBlock(ItemChisel.java:326) ~[ItemChisel.class:?]
	at mod.chiselsandbits.network.packets.PacketChisel.doAction(PacketChisel.java:190) ~[PacketChisel.class:?]
	at mod.chiselsandbits.network.packets.PacketChisel.server(PacketChisel.java:92) ~[PacketChisel.class:?]
	at mod.chiselsandbits.network.ModPacket.func_148833_a(ModPacket.java:57) ~[ModPacket.class:?]
	at org.spongepowered.common.network.PacketUtil.onProcessPacket(PacketUtil.java:92) ~[PacketUtil.class:1.12.2-2705-7.1.0-BETA-3422]
	at net.minecraft.network.PacketThreadUtil$1.redirect$onProcessPacket$zle000(SourceFile:539) ~[hv$1.class:?]
	at net.minecraft.network.PacketThreadUtil$1.run(SourceFile:13) ~[hv$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[?:1.8.0_181]
	at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:1.8.0_181]
	at net.minecraft.util.Util.func_181617_a(SourceFile:46) ~[h.class:?]
	... 7 more
```

**Comment by AlgorithmX2** [comment]
Still? that looks rather new to me... I've never seen it at least.

Looks like sponge is casting the world inappropriately. I don't think there is much I can do about that, without removing support for harvest materials.

Since that method takes a IBlockAccess assuming its a world is pretty poor form IMO.

**Comment by TeraNovell** [comment]
This Issue has been Fixed by: [#2420](https://github.com/SpongePowered/SpongeForge/pull/2420)

**Comment by AlgorithmX2** [comment]
Sounds good to me.
